### PR TITLE
EIP-1559 behavior polish

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-1926
Fixes RNBW-1927
Fixes RNBW-1931
Fixes RNBW-1935
Fixes RNBW-1948
Fixes RNBW-1930

### CODE REVIEW

File: src/components/gas/GasSpeedButton.js [hunk 1]
```diff
@@ -225,7 +225,7 @@ const GasSpeedButton = ({
       if (selectedSpeed === CUSTOM) {
         openCustomGasSheet();
         Keyboard.dismiss();
-        if (isEmpty(gasFeeParamsBySpeed[CUSTOM])) {
+        if (isEmpty(gasFeeParamsBySpeed[URGENT])) {
```
Comment (jinchung) [inline src/components/gas/GasSpeedButton.js:228]:
```if (isEmpty(gasFeeParamsBySpeed[CUSTOM])) {

File: src/components/expanded-state/custom-gas/FeesPanel.js [hunk 2]
```diff
@@ -386,6 +399,24 @@ export default function FeesPanel({
     [selectedOptionIsCustom]
   );
 
+  useEffect(() => {
+    const navigationRoutes = dangerouslyGetState().routes;
+    const lastRoute = navigationRoutes[navigationRoutes.length - 1]?.name;
```
Comment (jinchung) [inline src/components/expanded-state/custom-gas/FeesPanel.js:404]: ```const lastRoute = navigationRoutes?.[navigationRoutes.length - 1]?.name;```

**Review by jinchung — APPROVED** [review]
🍔

**Review by jinchung — DISMISSED** [review]
🍔

**Review by jinchung — DISMISSED** [review]
🍣

**Comment by estebanmino** [comment]
PoW https://recordit.co/b3BYNKIiQ1

**Review by jinchung — APPROVED** [review]
🍗

**Review by skylarbarrera — DISMISSED** [review]

**Comment by estebanmino** [comment]
fixes RNBW-1930

**Review by jinchung — DISMISSED** [review]
🔥

**Review by skylarbarrera — DISMISSED** [review]
:shipit:

**Comment by jinchung** [comment]
PoW: https://recordit.co/YksBtkCFJr

**Comment by christianbaroni** [comment]
Couple small issues still @jinchung is taking a look

**Comment by christianbaroni** [comment]
yo sorry @estebanmino !! I did not mean to close this

**Review by jinchung — APPROVED** [review]
Christian approves this too 🌮

**Review by christianbaroni — APPROVED** [review]
🌈

**Pull request merged**